source/utils/LogManager.h

```diff
@@ -18,45 +18,64 @@
+    #define OD_LOG_WRN(message)                      LogManager::getSingleton().logMessage(LogMessageLevel::WARNING, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+    // DEPRECATED: Use OD_TRACE_INF instead.
+    #define OD_LOG_INF(message)                      LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+    // DEPRECATED: Use OD_TRACE_DBG instead.
+    #define OD_LOG_DBG(message)                      LogManager::getSingleton().logMessage(LogMessageLevel::TRIVIAL, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+
+    // DEPRECATED: Use OD_ASSERT instead.
+    #define OD_ASSERT_TRUE(condition)                if (!(condition)) LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, "", __FILE__, __LINE__, #condition, 0)
+    // DEPRECATED: Use OD_ASSERT_MSG instead.
+    #define OD_ASSERT_TRUE_MSG(condition, message)   if (!(condition)) LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+#endif
+
+#define OD_TRACE_ERR(_module, _message, ...)         LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, #_module, __FILE__, __LINE__, _message, __VA_ARGS__)
+#define OD_TRACE_WRN(_module, _message, ...)         LogManager::getSingleton().logMessage(LogMessageLevel::WARNING, #_module, __FILE__, __LINE__, _message, __VA_ARGS__)
+#define OD_TRACE_INF(_module, _message, ...)         LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL, #_module, __FILE__, __LINE__, _message, __VA_ARGS__)
+#define OD_TRACE_DBG(_module, _message, ...)         LogManager::getSingleton().logMessage(LogMessageLevel::TRIVIAL, #_module, __FILE__, __LINE__, _message, __VA_ARGS__)
+
+#define OD_ASSERT(_condition)                        if (!(_condition)) LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, "Assert", __FILE__, __LINE__, #_condition, 0)
+#define OD_ASSERT_MSG(_condition, _message, ...)     if (!(_condition)) LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, "Assert", __FILE__, __LINE__, _message, __VA_ARGS__)
 
 //! \brief Helper/wrapper class to provide thread-safe logging when ogre is compiled without threads.
 class LogManager : public Ogre::Singleton<LogManager>
 {
 public:
-    LogManager() {};
-    //! \brief Log a message to the game log.
-    virtual void logMessage(const std::string& message, LogMessageLevel lml) = 0;
+    LogManager();
+    ~LogManager();
+
+    //! \brief Add a sink for log messages.
+    void addSink(const std::shared_ptr<LogSink>& sink);
 
-    //! \brief Set the log detail level.
-    //Messages lower than this level will be filtered.
-    virtual void setLogDetail(LogMessageLevel ll) = 0;
+    //! \brief Log a message to the sinks.
+    void logMessage(LogMessageLevel level, const char* module, const char* filepath, int line, const char* format, ...);
```

Comment:
Here, you have changed too much the API IMHO. The basic idea was to have debug logs that can be activated when we want to test something in particular.